### DISCUSSION TITLE
Relay language config

### DISCUSSION THREAD

**Discussion by artola** [vercel/next.js#35904] [Ideas]
### Describe the feature you'd like to request

The docs show how it is possible to use Next (@latest) with Relay (v13) and it works super!
https://nextjs.org/docs/advanced-features/compiler#relay

Nevertheless, Relay v14 (@master) has the possibility to emit types for different languages, before it was only possible `flow` and `typescript` but now has been added `javascript` language (no types emitted of course).

```
wait  - compiling...
error - ./node_modules/next/dist/client/dev/amp-dev.js
Error: failed to deserialize as next_swc::TransformOptions
JSON: {"jsc":{"parser":{"syntax":"ecmascript","dynamicImport":true,"decorators":false,"jsx":true,"importAssertions":true},"experimental":{"keepImportAssertions":true},"transform":{"legacyDecorator":false,"decoratorMetadata":false,"react":{"importSource":"react","runtime":"automatic","pragm

// ...

Caused by:
    unknown variant `javascript`, expected `typescript` or `flow` at line 1 column 981
```

### Describe the solution you'd like

Could be possible add to the config parser for `language` also `javascript`?
Mainly in the **canary** version would be fine.

### Describe alternatives you've considered

Waiting for next Relay release would take too long while we try to test any new feature in the whole stack.
May be @alunyov could give us some ETA for Relay, or even better confirm that `javascript` configuration option is stable and ongoing to be part of next release.

**Comment by alunyov** [comment]
> could give us some ETA for Relay, or even better confirm that javascript configuration option is stable

- We're planning on releasing v14 in a few weeks. 
- `javascript` option will be in the v14. 
 
As a workaround for `Next.js` config you can still use `flow`. flow-types are generated with comments, so Relay's generated artifacts are still valid JS.

**Reply by artola** [reply]
@alunyov Thanks for the info. Right, I used even on v13 the `flow` language as typings are just comments.

I am preparing a workshop for the [NDC Conferences](https://ndcporto.com/) and trying to show the "future" versions of the whole stack (aka `@master`, `@canary`, `@next`).

Last week I presented the same workshop in Zürich using flow, just now reviewing the latest changes I stumbled in this new language option.

**Comment by artola** [comment]
@alunyov @timneutkens This is still open, even after release of Relay v14.1
With the actual setup, if I pass `language: 'javascript'` ... it still throws ``unknown variant `javascript`, expected `typescript` or `flow``.

**Reply by artola** [reply]
@timneutkens Still this bug/limitation persist on `12.3.2-canary.13`. Is it only because of the valid config recognised by next?

**Reply by artola** [reply]
Still a limitation on `Next.js v13.0.3-canary.2`.

**Reply by timneutkens** [reply]
Feel free to open a PR upgrading the Relay package in the transform!

**Reply by artola** [reply]
> Feel free to open a PR upgrading the Relay package in the transform!

Could you give me some hint where to look? For example:

https://github.com/vercel/next.js/blob/canary/packages/next-swc/crates/core/src/relay.rs

**Comment by artola** [comment]
Implemented by: https://github.com/vercel/next.js/commit/9c2c8587be77c176216470a7789f9865e8c326b8
